integrations/java/iceberg-1.2/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseTableOperations.java

```diff
@@ -186,6 +187,18 @@ protected CreateUpdateTableRequestBody constructMetadataRequestBody(
         && metadata.properties().containsKey(OPENHOUSE_TABLE_TYPE_KEY)) {
       createUpdateTableRequestBody.setTableType(getTableType(base, metadata));
     }
+    // TODO: consider allowing this for any table type, not just replica tables
+    if (isMultiSchemaUpdateCommit(base, metadata)
+        && getTableType(base, metadata)
+            == CreateUpdateTableRequestBody.TableTypeEnum.REPLICA_TABLE) {
+      Map<String, String> intermediateSchemas = new HashMap<>();
+      int startSchemaId = base == null ? 0 : base.currentSchemaId() + 1;
+      for (int i = startSchemaId; i < metadata.currentSchemaId(); i++) {
+        intermediateSchemas.put(
+            String.valueOf(i), SchemaParser.toJson(metadata.schemasById().get(i), false));
+      }
+      createUpdateTableRequestBody.setIntermediateSchemas(intermediateSchemas);
```

Comment:
So we are determining intermediate schema here and this method is called as part of put snapshots and create/update table.